### ISSUE TITLE
Support for integer interpolation (torch.nn.functional.interpolate)

### ISSUE THREAD

**Issue by martinruenz** [pytorch/pytorch#13218]
## 🚀 Feature
Currently, `torch.nn.functional.interpolate` does not support integer formats (I tested int and uint8). I would appreciate having support for these formats, or a description in the documentation mentioning why they are not supported. 

## Motivation

Consider having an image representing a mask. To scale this image (nearest neighbor) using pytorch I would need to use the following hack:
```python
mask = (torch.randn((1,1,50,50)) > 1)   
torch.nn.functional.interpolate(mask.float(),scale_factor=0.5).int()  
```

This is introducing unnecessary casts and less clear compared to the following code, which fails:
```python
mask = (torch.randn((1,1,50,50)) > 1)   
torch.nn.functional.interpolate(mask,scale_factor=0.5) # RuntimeError: upsample_nearest2d_forward is not implemented for type torch.ByteTensor

```

**Comment by vadimkantorov** [comment]
(was also tracked at https://github.com/pytorch/pytorch/issues/5580)

**Comment by vadimkantorov** [comment]
Bool support would also be useful (for masks)

**Comment by fmassa** [comment]
Agreed, it would be great to extend interpolate to other types than float
